src/Control/Monad/Aff/Internal.js

```diff
@@ -25,6 +25,22 @@ exports._takeVar = function (nonCanceler, avar) {
   };
 };
 
+exports._tryTakeVar = function (nothing, just, nonCanceler, avar) {
+  return function (success, error) {
+    if (avar.error !== undefined) {
+      error(avar.error);
+    } else if (avar.producers.length > 0) {
+      avar.producers.shift()(function (x) {
+        success(just(x));
+        return nonCanceler;
```

Comment:
updated, i just using the style i see on FFI file for Aff.js, there ```_pure```, ```_throwError``` return nonCanceller. But, i guess you are right, because ```_fmap``` in that file even doesn't return nonCanceller, nor return the return value